spec/ics-25-handler-interface/README.md

```diff
@@ -0,0 +1,314 @@
+
+```coffeescript
+function queryConnection(string id) -> Maybe<ConnectionInfo>
+```
+
+`initCloseConnection` initiates the graceful connection closing process. It will fail if there are any open channels using the connection.
+
+```coffeescript
+function initCloseConnection(string id) -> Maybe<Err>
+```
+
+`tryCloseConnection` continues the graceful connection closing process. It will fail if there are any open channels using the connection.
+
+```coffeescript
+function tryCloseConnection(identifier, proofInit) -> Maybe<Err>
+```
+
+`ackCloseConnection` finalizes the graceful connection closing process.
+
+```coffeescript
+function ackCloseConnection(identifier, proofTry) -> Maybe<Err>
+```
+
+`equivocationCloseConnection` force-closes a connection upon discovery of an equivocation. The equivocation must first be submitted to the associated client.
+
+```coffeescript
+function equivocationCloseConnection(identifier) -> Maybe<Err>
+```
+
+Implementations of `initConnection`, `tryConnection`, `ackConnection`, `confirmConnection`, `queryConnection`, `initCloseConnection`, `tryCloseConnection`, `ackCloseConnection`, and `equivocationCloseConnection` are defined in ICS 3.
+
+### Channels
+
+By default, channels are owned by the creating module, meaning only the creating module can inspect, close, or send on the channel.
+
+`ChannelOptions` contains all the parameter choices required to create a new channel.
+
+```golang
+type ChannelOptions struct {
+  string          connectionIdentifier
+  bool            ordered
+}
+```
+
+`ChannelInfo` contains metadata about an existing channel.
+
+```golang
+type ChannelInfo struct {
+  ChannelOptions options
+}
+```
+
+`createChannel` tries to create a new channel with the provided options, failing if the connection is not found or the options are invalid, returning a unique allocated identifier if successful.
+
+```coffeescript
+function createChannel(ChannelOptions options) -> string
+```
+
+`queryChannel` queries an existing channel by known identifier, returning the associated metadata if found.
+
+```coffeescript
+function queryChannel(string id) -> Maybe<ChannelInfo>
+```
+
+`closeChannel` initiates the graceful channel closing process as defined in ICS 4.
+
+```coffeescript
+function closeChannel(string id) -> Future<Maybe<Err>>
+```
+
+Implementations of `createChannel`, `queryChannel`, and `closeChannel` are defined in ICS 4.
+
+### Packets
+
+Packets are permissioned by channel (only a module which owns a channel can send on it).
+
+`sendPacket` attempts to send a packet, returning an error if the packet cannot be sent (perhaps because the sending module does not own the channel in question), and returning a unique identifier if successful.
+
+The returned identifier will be the same as that sent by the timeout handler, so it can be used by the sending module to associate a specific action with a specific packet timeout.
```

Comment:
What timeout handler is this? This sentence needs a bit of clarification IMO.

Reply:
`timeoutPacket`. The idea is that the module could associate a specific action with a timeout of a packet once it has sent the packet (without explicitly passing function pointers as callbacks).

Clarified a bit.

Reply:
Also see https://github.com/cosmos/ics/issues/86